File: C++/christmas-countdown.cpp
#include <iostream>
#include <ctime>
#include <string>
#include <sstream>
#include <iomanip>
#include <Windows.h>

using namespace std;

int main()
{
    while (true) {
        time_t now = time(0);
        tm *ltm = localtime(&now);
        int year = 1900 + ltm->tm_year;
        int month = 11;
        int day = 25;
        tm christmas = {0, 0, 0, day, month, year - 1900};
        time_t xmas = mktime(&christmas);
        double difference = difftime(xmas, now);
        int days = difference / (60 * 60 * 24);
        difference -= days * (60 * 60 * 24);
        int hours = difference / (60 * 60);
        difference -= hours * (60 * 60);
        int minutes = difference / 60;
        difference -= minutes * 60;
        int seconds = difference;
        system("cls");
        cout << "Until Christmas 2022\n" << days << " days, " << hours << " hours, " << minutes << " minutes, and " << seconds << " seconds" << endl;
        Sleep(1000);
    }
}

File: Python/christmas-countdown.py
import time
import datetime
import os

def countdown(stop):
    while True:
        difference = stop - datetime.datetime.now()
        count_hours, rem = divmod(difference.seconds, 3600)
        count_minutes, count_seconds = divmod(rem, 60)
        count_milliseconds = difference.microseconds / 1000
        if difference.days == 0 and count_hours == 0 and count_minutes == 0 and count_seconds == 0:
            print("Merry Christmas!")
            break
        else:
            os.system('cls')
            print('Until christmas there is:' + "\n" + str(difference.days) + ' days, ' + str(count_hours) +
                  ' hours, ' + str(count_minutes) + ' minutes, ' + str(count_seconds) + ' seconds, ' + str(count_milliseconds) + ' milliseconds')
            time.sleep(0.001)

if __name__ == '__main__':
    christmas = datetime.datetime(2022, 12, 25)
    countdown(christmas)